### PR TITLE
ridgeline moved and render js refactored

### PR DESCRIPTION
**Pull request by plotthatstory**

### CODE REVIEW

File: src/charts/ridgeline/validateAndRender.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  checkColorArray,
+} from '../../utils/validation/optionValidations'
+import {
+  showErrors,
+  validateColumnsWithDimensions,
+} from '../../utils/validation/validations'
+import { renderChart } from './render'
+
+const dimensionTypes = {
+  xField: [],
```
Comment (shreshthmohan) [inline src/charts/ridgeline/validateAndRender.js:23]: shouldNotBeBlank validation?

File: demo/src/ridgeline/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  overlap: 7,
+  yValueFormatter: '.1f',
+  yValuePrefix: '',
+  yValuePostfix: 'M',
+
+  /* seriesField */
+  seriesLabelPosition: 'left', // ['left', 'right']
+
+  /* colorField */
+  colorRange: d3.schemeTableau10,
```
Comment (shreshthmohan) [inline demo/src/ridgeline/index.js:29]: fix eslint error. add d3 to global (see first line)

File: src/charts/ridgeline/render.js [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+  applyInteractionStyles({ activeOpacity, inactiveOpacity })
+
+  const coreChartWidth = 1000
+  const {
+    svg,
+    coreChartHeight,
+    allComponents,
+    chartCore,
+    widgetsLeft,
+    widgetsRight,
```
Comment (shreshthmohan) [inline src/charts/ridgeline/render.js:484]: fix eslint error (declared but never used)

**Review by shreshthmohan — CHANGES_REQUESTED** [review]

File: src/charts/ridgeline/validateAndRender.js (hunk 1 above)
Comment (plotthatstory) [reply src/charts/ridgeline/validateAndRender.js:23]: I left it blank intentionally but cannot remember why. Probably a mistake. Made it shouldNotBeBlank

File: demo/src/ridgeline/index.js (hunk 2 above)
Comment (plotthatstory) [reply demo/src/ridgeline/index.js:29]: Fixed
`/* global viz, d3 */`

File: src/charts/ridgeline/render.js (hunk 3 above)
Comment (plotthatstory) [reply src/charts/ridgeline/render.js:484]: Still returning widgets right from setupChartArea fucntion. Removed it when the function is called from renderChart.
We can move the setupCartArea function to general utils.

**Pull request merged**